Fix the ElasticaDatagridBuilder when use autocomplete filter

chervaliery opened the pull request:
<!-- THE PR TEMPLATE IS NOT AN OPTION. DO NOT DELETE IT! -->

<!--
    Specify which issues will be fixed/closed.
    Remove it if this is not related.
-->

Closes #24 
### Changelog

<!-- MANDATORY
    Fill the changelog part inside the code block.
    Remove unneeded sections.
    Follow this schema: http://keepachangelog.com/
-->

``` markdown
### Fixed
 - `ElasticaDatagridBuilder` now handles autocomplete filter
```
### Todo
- [x] separate commits
- [ ] with meaningful messages
### Subject

Fixe the ElasticaDatagridBuilder and the DependencyInjection/Configuration according to the discussion in #24 

<!-- Describe your Pull Request content here -->

greg0ire commented:
Thanks @chervaliery ! Can you read and apply this though : https://github.com/sonata-project/SonataAdminSearchBundle/blob/master/CONTRIBUTING.md#the-commit-message ?

greg0ire commented on Builder/ElasticaDatagridBuilder.php:
```diff
@@ -131,10 +131,10 @@ public function isSmart(AdminInterface $admin, array $values = array())
-        $fullCurrentAction = $admin->getRequest()->attributes->get('_controller');
-        $currentAction = explode('::', $fullCurrentAction);
+        $logicalControllerName = $admin->getRequest()->attributes->get('_controller');
+        $currentAction = explode(':', $logicalControllerName);
```
This should be in a separate commit (same PR is fine)

greg0ire commented on DependencyInjection/Configuration.php:
```diff
@@ -33,7 +33,7 @@ public function getConfigTreeBuilder()
                           ->scalarNode('finder')->isRequired()->end()
                           ->arrayNode('actions')
-                              ->defaultValue(array('list'))
+                              ->defaultValue(array('list', 'retrieveAutocompleteItems'))
```
This should also be in a separate commit

Commit:
Fix the retrieveAutocompleteItems extract from controller

Work's now for
FrameworkBundle:Redirect:redirect
sonata.admin.controller.admin:retrieveAutocompleteItemsAction
sonata.admin.controller.admin::retrieveAutocompleteItemsAction

Commit:
Change fullCurrentAction variable name to logicalControllerName

It's to use the same terminology as Symfony

Commit:
Add value to finder service for autocomplete

Add the retrieveAutocompleteItems value to the default admin_finder_services

chervaliery commented:
It should be good this time

greg0ire commented:
It's way better like this, but the third commit message looks strange : 
What do you mean by `admin_finder_services` ?

Also, make sure you associate your excellium mail address to your account if you want your contributions to show up in your account ;)

chervaliery commented:
admin_finder_services is the name of the node in the configuration tree.
I don't know how to explain it more clearly.

core23 commented:
Please change your PR title. I can't get what this PR is about when browsing the PR list. Same for the changelog.

chervaliery commented:
I rename it

core23 commented:
```
### Fixed
ElasticaDatagridBuilder
DependencyInjection/Configuration
```

WHAT did you fix?

chervaliery commented:
Is it better ?

chervaliery commented:
Is there something else to do ?

greg0ire commented:
> admin_finder_services is the name of the node in the configuration tree.
> I don't know how to explain it more clearly.

Oh ok, I did not understand that it was supposed to be part of the same sentence.

Try to keep the commit message subject < 50 characters, like I asked, and I will merge this. See how your sentence is broken in the middle [here](https://github.com/sonata-project/SonataAdminSearchBundle/pull/41/commits/0fddf6d6a08756713364533be2c3df589ca89321) ? This is what you should try to avoid.

So

```
Small message under fifty characters

Empty line followed by a § that explains in detail what you are doing.
```

chervaliery commented:
I change the commit message and title

greg0ire commented:
Great job! Let's wait for the build to complete!

greg0ire commented:
Thanks again, nice job @chervaliery !
